sound/soc/sof/topology.c

```diff
@@ -751,13 +751,15 @@ static const struct sof_topology_token led_tokens[] = {
 	 get_token_u32, offsetof(struct snd_sof_led_control, direction), 0},
 };
 
```

Comment:
Commit message is quite vague.

I mentioned before that it's not clear if you only keep the last one, or override the rest.

there probably a typo (need to know).

'the amount of combination items' is convoluted. Is this the number of tuples in the array? what is a combination anyway? Did you mean array?

what is 'destination'?

words matter if you want people to get the point, and what's different from your previous PR

Reply:
yes I have to say the concepts are confusing to me also. So what we are doing in every parsing is to transform from alsa topology "arrays" to ipc messages (destination). Now it seems that every same type parameters are in 1 alsa array, so if we have like 2 integers and 1 string, we are trying to parse some kind of combination of arrays (not 1 array) to ipc struct.

So currently if you have several this kind of combinations following each other, and you call sof_parse_token, it will parse every combination on top of the destination. So only the last one will remain in destination, others are lost. So exactly we should have the "sof_parse_token_arrays" or "sof_parse_token_combinations" what you suggested.

I'll try to somehow make the message better...

Anyway